anndata/tests/helpers.py

```diff
@@ -442,13 +442,16 @@ def assert_equal_h5py_dataset(a, b, exact=False, elem_name=None):
 
 @assert_equal.register(DaskArray)
 def assert_equal_dask_array(a, b, exact=False, elem_name=None):
-    from dask.array.utils import assert_eq
-
-    if exact:
-        assert_eq(a, b, check_dtype=True, check_type=True, check_graph=False)
-    else:
-        # TODO: Why does it fail when check_graph=True
-        assert_eq(a, b, check_dtype=False, check_type=False, check_graph=False)
+    assert_equal(b, a.compute(), exact, elem_name)
+    # TODO: Figure out why we did this
+    # from dask.array.utils import assert_eq
+    #
+    # I believe the above fails for sparse matrices due to some coercion to np.matrix
+    # if exact:
+    #     assert_eq(a, b, check_dtype=True, check_type=True, check_graph=False)
+    # else:
+    #     # TODO: Why does it fail when check_graph=True
+    #     assert_eq(a, b, check_dtype=False, check_type=False, check_graph=False)
```

Comment:
I remember not wanting to use compute for the equality check to see if the lazy version works fine with the assertion. However, it would also check if the computation graph was also the same, which turned out to fail. Because lazy equality check should also be dasks feature I think we can maybe write a assert_eq function for two dask arrays and compute if other isn't a dask array.

Reply:
I think dask's `assert_eq` always computes the values. To me, this makes sense for a test assertion, since it's okay if it's a bit slow.

* `assert_eq` definition: https://github.com/dask/dask/blob/b015ea8bf87f4c525379719f8a1cf2851c55bb58/dask/array/utils.py#L284
* `_get_dt_meta_computed` inner function definition: https://github.com/dask/dask/blob/b015ea8bf87f4c525379719f8a1cf2851c55bb58/dask/array/utils.py#L245

Just graph based comparisons are probably a better fit for the `anndata._core.merge.equals`.